scripts/generate.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+[[ "${1:-}" != "--help" ]] || {
+  cat <<-EOF
+	Gets .did files and generates canister bindings from them.
+
+	Note: This does so WITHOUT deploying any canisters.  Typically "dfx generate"
+	requires all the canisters to be deployed locally, which seems absurdly heavyweight.
+	This code just downloads the candid files and puts them where they would normally
+	be found for a local deployment.  Much faster!
+	EOF
+  exit 0
+}
+
+# Gets all .did files listed in dfx.json.
+#
+# .did files are placed in the local .dfx directory, where the .did files are expected by `dfx generate` and other commands.
+function install_did_files() {
+  jq -r '.canisters | to_entries | .[] | select(.value.candid != null) | "\(.key) \(.value.candid)"' dfx.json |
+    while read -r line; do
+      IFS=', ' read -r -a array <<<"$line"
+      canister_name="${array[0]}"
+      source="${array[1]}"
+      destination=".dfx/local/canisters/${array[0]}/${array[0]}.did"
+      mkdir -p "$(dirname "$destination")"
+      case "$source" in
+      http*) curl -sSL "$source" >"$destination" ;;
+      *) if test -e "$source"; then cp "$source" "$destination"; else echo "WARNING: $canister_name did file not found at $source"; fi ;;
+      esac
+    done
+}
+
+# Create local .did files
+./scripts/did.sh
+# Download .did files without an explicit url in dfx.json
+./scripts/download.icp.sh
```

Comment:
We probably don't require download icp, ckbtc and cketh as we do no generate their types. Any particular reason why you download those here?

Reply:
Running a blanket `dfx generate`, as in the original `package.json`, will fail if some candid files are missing, like this:
```
Generating type declarations for canister icp_ledger:
Error: Failed while trying to generate type declarations for 'icp_ledger'.
Caused by: Failed while trying to generate type declarations for 'icp_ledger'.
  Candid file: /home/max/dfn/oisy-wallet/branches/check-generate-in-ci/target/ic/icp_ledger.did doesn't exist.
```
I guess one could generate only the canisters for which there are directories in `declarations`, like this:

```
for canister in $(ls src/declarations/) ; do echo "Generating bindings for $canister" ; dfx generate "$canister" ; done
```
Is that what you mean?  I think it's a good idea.  Less files to download => faster execution. :-)

Reply:
I did not knew. I think we should just get rid of `dfx generate` and directly use `candid-extractor`.

That's what I do in Juno. e.g. https://github.com/junobuild/juno/blob/main/scripts/did.sh

I can provide a PR to do so once we have merged this PR.